### PR TITLE
feat: add query params to request context

### PR DESCRIPTION
**Pull request by sauyon**
See #3674.

### CODE REVIEW

File: src/bentoml/_internal/context.py [hunk 1]
```diff
@@ -92,16 +92,21 @@ def from_http(request: "starlette.requests.Request") -> "InferenceApiContext":
     class RequestContext:
         metadata: Metadata
         headers: Metadata
+        query_params: Metadata
 
-        def __init__(self, metadata: Metadata):
+        def __init__(self, metadata: Metadata, query_params: Metadata):
             self.metadata = metadata
             self.headers = metadata
+            self.query_params = metadata
```
Comment (Lorenzo-N) [inline src/bentoml/_internal/context.py:100]: This line should be `self.query_params = query_params`

Comment (parano) [inline src/bentoml/_internal/context.py:100]:
```suggestion
            self.query_params = query_params
```

Comment (sauyon) [reply src/bentoml/_internal/context.py:100]: Indeed :facepalm:. Adding a test for this.

**Comment by sauyon** [comment]
Sorry this took a while, was hoping to get #3696 in so I could use it for the tests, but clearly that was somewhat ambitious.

**Comment by parano** [comment]
LGTM!

**Review by aarnphm — APPROVED** [review]

**Pull request merged**